### PR TITLE
Improve timed token security

### PR DESCRIPTION
**Pull request by JeffAbrahamson**
Our timed tokens were emminently forgeable, although it seemed quite a
lot of work for an adversary to do so for the very little gain.  This
commit addresses that issue by upping (significantly) the amount of
work necessary to forge a timed token.  We believe the difficulty of
forging a timed token is now on a par with the difficulty of breaking
SHA1 or SHA256.

This increases the length of our timed tokens, but we don't currently
consider that to be a problem.

### CODE REVIEW

File: transport_nantes/asso_tn/utils.py [hunk 1]
```diff
@@ -56,8 +70,9 @@ def make_timed_token(string_key, minutes, int_key=0, test_value_now=None):
     int_key_str = int_to_base36(int(int_key))
     hmac = salted_hmac(soon_seconds + int_key_str, rand_value + str(string_key)
                        ).hexdigest()[:20]
-    token = '{rnd}|{e}|{t}|{p}|{h}'.format(
+    token = '{rnd}|{hr}|{e}|{t}|{p}|{h}'.format(
```
Comment (Shriukan33) [inline transport_nantes/asso_tn/utils.py:73]: Adding another element to the token will raise a ValueError when other functions using this token will unpack the resulting token.

Comment (JeffAbrahamson) [reply transport_nantes/asso_tn/utils.py:73]: Isn't the only function to do that `token_validate`?

Comment (JeffAbrahamson) [reply transport_nantes/asso_tn/utils.py:73]: Yes, there's a race condition.  All those pending tokens out there...  ;-)

Comment (JeffAbrahamson) [reply transport_nantes/asso_tn/utils.py:73]: Though that's a reminder that we need to make sure we're always robust against a bad token coming in, offering something reasonably logical to help the user recover.

Comment (Shriukan33) [reply transport_nantes/asso_tn/utils.py:73]: The token_valid function will return a variable number of values, one for each value seperated by a pipe.
And we need to store the values out of token_valid, to extract the email, and send_record_id if any.

Comment (JeffAbrahamson) [reply transport_nantes/asso_tn/utils.py:73]: The job of `token_valid()` is to extract the two values we put in.  No other entity should be trying to parse the token and expecting not to be fragile.

Can you construct a scenario and show me in code how you could call `make_timed_token()` and then pass the token to `valid_token()` and have the string and int that you recover not be what you put in?

**Comment by Shriukan33** [comment]
LGTM

**Review by Shriukan33 — APPROVED** [review]

**Pull request merged**